src/Kobo/SystemDialog.cpp

```diff
@@ -224,6 +244,33 @@ SystemWidget::UpdateBacklightButtons(int percent)
   }
 }
 
+inline void
+SystemWidget::IncreaseBacklightColourTemperature()
+{
+  int current_colour = KoboGetBacklightColour();
+  KoboSetBacklightColour(current_colour + 2);
+  UpdateBacklightColourButtons(current_colour + 2);
+}
+
+inline void
+SystemWidget::DecreaseBacklightColourTemperature()
+{
+  int current_colour = KoboGetBacklightColour();
+  KoboSetBacklightColour(current_colour - 2);
+  UpdateBacklightColourButtons(current_colour - 2);
+}
+
+inline void
+SystemWidget::UpdateBacklightColourButtons(int colour)
+{
+  if(decrease_backlight_colour_temperature != nullptr) {
+    decrease_backlight_colour_temperature->SetEnabled(colour != 0);
```

Comment:
This enables the button if the color is negative

Reply:
Yes.  Good point.  This has been changed to be colour > 0

Reply:
No. You made a mistake: you made the colour a *signed* integer. Why did you choose to make it signed, why did you choose to allow negative numbers?